### PR TITLE
feat(dlm): supports viewing schedule task details

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
During the execution of the data archiving task, it is not possible to view execution details, making it difficult for users to understand the progress of the task. This PR provides the functionality to view the execution details.
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_4__add_schedule_task_unit.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  `schedule_task_id` bigint NOT NULL COMMENT 'ID of the related schedule task',
+  `type` varchar(120) NOT NULL COMMENT 'Task type',
+  `task_unit_parameters` text NOT NULL COMMENT 'Task parameters',
+  `execution_detail` text NOT NULL COMMENT 'Task execution detail',
+  `start_time` date NULL COMMENT 'Task start time',
+  `end_time` date NULL COMMENT 'Task end time',
+  `status` varchar(120) NOT NULL COMMENT 'Task status',
+  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'Record insertion time',
+  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'Record modification time',
+  INDEX `idx_schedule_task_unit_schedule_task_id`  (`schedule_task_id`)
```
[the remark this answers is not in the document]
Comment (guowl3) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_4__add_schedule_task_unit.sql]: Thanks for your advice. I have added a primary key to the table.

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleController.java [hunk 2]
```diff
@@ -92,6 +92,13 @@ public SuccessResponse<ScheduleDetailResp> detail(@PathVariable Long id) {
         return Responses.single(scheduleService.getById(id));
     }
 
+    @RequestMapping(value = "/scheduleConfigs/{scheduleId:[\\d]+}/scheduleTask/{scheduleTaskId:[\\d]+}",
+            method = RequestMethod.GET)
+    public SuccessResponse<ScheduleTaskResp> detailScheduleTask(@PathVariable Long scheduleId,
```
[the remark this answers is not in the document]
Comment (guowl3) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleController.java:97]: The detail method is used to view the details of a Schedule, while the detailScheduleTask method is used to view the details of the execution record.

[the remark this answers is not in the document]
Comment (yhilmare) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleController.java:97]: you may consider this: I am green hand and I know nothing about this, I notice that there are two method: `default` and `defaultScheduleTask`, how can I know the differences between these two methods?

I can change another way: why shouldn't the method `detail`'s response include the  execution record?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleTaskService.java [hunk 3]
```diff
@@ -62,14 +67,29 @@ public class ScheduleTaskService {
...
 
+    private final ScheduleTaskMapper scheduleTaskMapper = ScheduleTaskMapper.INSTANCE;
+
     @Value("${odc.log.directory:./log}")
     private String logDirectory;
 
     private static final String LOG_PATH_PATTERN = "%s/scheduleTask/%s-%s/%s/log.%s";
 
+    public ScheduleTaskResp detail(Long id) {
+        ScheduleTaskEntity entity = nullSafeGetById(id);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleTaskService.java:84]: `ScheduleService#nullSafeGetByIdWithCheckPermission` returns the `ScheduleEntity`

`ScheduleService#detailScheduleTask` call the method `ScheduleService#nullSafeGetByIdWithCheckPermission` and ignore the return value and call `nullSafeGetById`, why should we query schedule entity twice?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * @Author：tinker
+ * @Date: 2024/5/9 09:39
+ * @Descripition:
+ */
+public class DataArchiveExecutionDetail implements ExecutionDetail {
+
+    private String tableName;
+
+    private String type;
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java:29]: why string?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    private Long processRowCount;
+
+    private Long readRowCount;
+
+    private String status;
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java:35]: why string?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ * @Date: 2024/5/9 09:39
+ * @Descripition:
+ */
+public class DataArchiveExecutionDetail implements ExecutionDetail {
+
+    private String tableName;
+
+    private String type;
+
+    private Long processRowCount;
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java:31]: processedRowCount should be better?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskShard.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    private TaskShardParameters taskParameters;
+
+    private ExecutionDetail executionDetail;
+
+    private Date startTime;
+
+    private Date endTime;
+
+    private String status;
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskShard.java:44]: why string?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/TaskShardParameters.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.schedule.model;
+
+/**
+ * @Author：tinker
+ * @Date: 2024/5/10 11:12
+ * @Descripition:
+ */
+public interface TaskShardParameters {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/TaskShardParameters.java:23]: why should we need this interface?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ExecutionDetail.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.schedule.model;
+
+/**
+ * @Author：tinker
+ * @Date: 2024/5/10 10:53
+ * @Descripition:
+ */
+public interface ExecutionDetail {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ExecutionDetail.java:23]: why should we need this interface?

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_4__add_schedule_task_shard.sql [hunk 10]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS `schedule_task_shard` (
+  `id` bigint AUTO_INCREMENT  NOT NULL COMMENT 'ID of the schedule task unit',
+  `schedule_task_id` bigint NOT NULL COMMENT 'ID of the related schedule task',
+  `type` varchar(120) NOT NULL COMMENT 'Task type',
+  `task_parameters` text NOT NULL COMMENT 'Task parameters',
+  `execution_detail` text NOT NULL COMMENT 'Task execution detail',
+  `start_time` date NULL COMMENT 'Task start time',
```
[the remark this answers is not in the document]
Comment (guowl3) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_4__add_schedule_task_shard.sql]: I have changed the field type of date to datetime.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java (hunk 4 above)
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java]: It has been changed to an enum type.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java (hunk 5 above)
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/DataArchiveExecutionDetail.java:33]: It has been changed to an enum type.

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (guowl3) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/ScheduleController.java:97]: I renamed detail to detailSchedule, please note that Schedule and ScheduleTask are two different objects.

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql [hunk 11]
```diff
@@ -0,0 +1,31 @@
+CREATE TABLE IF NOT EXISTS `dlm_job` (
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql:1]: concept may confused with odc task frameowork, may `dlm_table_unit`

Comment (guowl3) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql]: renamed to dlm_table_unit

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  `id` bigint AUTO_INCREMENT NOT NULL COMMENT 'ID of the dlm job object',
+  `schedule_task_id` bigint NOT NULL COMMENT 'ID of the related schedule task, refer to schedule_task.id',
+  `dlm_job_id` varchar(120) NOT NULL COMMENT 'Unique identifier of the dlm job object',
+  `table_name` varchar(120) NOT NULL COMMENT 'Name of the source table',
+  `fire_time` varchar(120) NOT NULL COMMENT 'The actual time the trigger fired',
+  `target_table_name` varchar(120) NULL COMMENT 'Name of the target table',
+  `source_datasource_info` bigint NOT NULL COMMENT 'Info of the source datasource',
```
Comment (krihy) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql]: it is hard understand `source_datasource_info` with bigint

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+  `target_datasource_info` bigint NULL COMMENT 'Info of the target datasource',
```
Comment (krihy) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql]: as see above

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+INDEX `idx_dlm_job_schedule_task_id` (`schedule_task_id`),
+CONSTRAINT `pk_dlm_job_id` PRIMARY KEY (`id`),
+CONSTRAINT `uk_dlm_job_dlm_job_id` UNIQUE (`dlm_job_id`)
+);
+
+CREATE TABLE IF NOT EXISTS `dlm_job_statistic` (
+  `id` bigint AUTO_INCREMENT NOT NULL COMMENT 'ID of the dlm job statistic object',
+  `dlm_job_id` varchar(120) NOT NULL COMMENT 'The dlm job ID that this statistic is associated with',
+  `processed_row_count` bigint NULL COMMENT 'Total number of rows processed by the dlm job',
+  `processed_rows_per_second` bigint NULL COMMENT 'Rate of processing in rows per second',
```
Comment (krihy) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql]: processed_rows_per_second and other per_second is dynamic value, what is frequency to save them

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/DLMJobStore.java [hunk 15]
```diff
@@ -65,6 +72,84 @@ public void destroy() {
...
+
+        } catch (SQLException e) {
+            log.warn("Update dlm job status failed,dlmTableUnitId={}", dlmTableUnitId, e);
+            return false;
+        }
+    }
+
+
+    public void storeDlmTableUnit(List<DlmTableUnit> dlmTableUnits) throws SQLException {
+        String sql = "INSERT INTO dlm_job (schedule_task_id, dlm_job_id, table_name, fire_time, " +
```
Comment (krihy) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/DLMJobStore.java:124]: may be `dlm_job_unit` more suitable

**Review by yizhouxw — DISMISSED** [review]
lgtm

**Review by yhilmare — DISMISSED** [review]
lgtm

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql (hunk 12 above)
Comment (guowl3) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql]: corrected to varchar

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql (hunk 14 above)
Comment (guowl3) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_0_5__add_dlm_job.sql]: table has been deleted

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/DLMJobStore.java (hunk 15 above)
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/dlm/DLMJobStore.java]: easily confused with Job

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Review by krihy — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**